connection.c

```diff
@@ -127,13 +128,63 @@ void ConnAsyncTerminate(Connection *conn)
     conn->flags |= CONN_TERMINATING;
 }
 
+static void handleAuth(redisAsyncContext *c, void *r, void *privdata)
+{
+    Connection *conn = privdata;
+    JoinLinkState *state = ConnGetPrivateData(conn);
+
+    redisReply *reply = r;
+    if (!reply) {
+        LOG_ERROR("Redis connection authentication failed: connection died");
+        ConnMarkDisconnected(conn);
+    } else if (reply->type == REDIS_REPLY_ERROR) {
+        LOG_ERROR("Redis connection authentication failed: %s", reply->str);
+        state->failed = true;
+        ConnMarkDisconnected(conn);
+    } else if (reply->type != REDIS_REPLY_STATUS || strcmp("OK", reply->str) != 0) {
+        LOG_ERROR("Redis connection authentication failed: unexpected response");
+        state->failed = true;
+        ConnMarkDisconnected(conn);
+    }
+
+    conn->state = CONN_CONNECTED;
+    conn->connect_oks++;
+    conn->last_connected_time = RedisModule_Milliseconds();
+
+    if (conn->connect_callback) {
+        conn->connect_callback(conn);
+    }
+}
+
+static void doAuthentication(Connection *conn, int status)
+{
+    RedisRaftCtx *rr = ConnGetRedisRaftCtx(conn);
+
+    if (status == REDIS_OK) {
+        if (redisAsyncCommand(ConnGetRedisCtx(conn), handleAuth, conn,
+                              "AUTH %s %s",
+                              rr->config->cluster_user ? rr->config->cluster_user : "default",
```

Comment:
If we always have username password,  you can just init in ConfigInit() like :
```c
config->cluster_username = RedisModuleStrdup("default");
config->cluster_password = RedisModuleStrdup("");
```

So, we'll avoid null checks here and above.

Reply:
so I actually did this originally, this was the error I was mentioning in slack about leak detector going of, hence why did this.  But I agree your would be better, will try and put it back and see what happens